tests/testthat/test-utils-read.R

```diff
@@ -1944,84 +1944,307 @@ test_that('readGrades properly retrieves the grades', {
+                         "shift": -0.02
+                         },
+                         {
+                         "inputValue": 2,
+                         "shift": 0
+                         }
+                         ]
+                         },
+                         {
+                         "periodOfApplicability": {
+                         "startTime": "2016-11-08T19:02:00.0000000Z",
+                         "endTime": "2017-04-13T02:00:00.0000000Z",
+                         "remarks": "Carried over from previous period. Ended when Rating 9.0 was started. JRC"
+                         },
+                         "shiftPoints": [
+                         {
+                         "inputValue": 1.4,
+                         "shift": -0.02
+                         },
+                         {
+                         "inputValue": 2,
+                         "shift": 0
+                         }
+                         ]
+                         }
+                         ],
+                         "baseRatingTable": [
+                         {
+                         "inputValue": 1.22,
+                         "outputValue": 0.54
+                         },
+                         {
+                         "inputValue": 1.831,
+                         "outputValue": 11.324
+                         },
+                         {
+                         "inputValue": 2.062,
+                         "outputValue": 19.302
+                         },
+                         {
+                         "inputValue": 2.368,
+                         "outputValue": 30.964
+                         },
+                         {
+                         "inputValue": 2.664,
+                         "outputValue": 42.756
+                         },
+                         {
+                         "inputValue": 3.28,
+                         "outputValue": 71.49
+                         },
+                         {
+                         "inputValue": 4.11,
+                         "outputValue": 114.36
+                         },
+                         {
+                         "inputValue": 5.13,
+                         "outputValue": 170.67
+                         },
+                         {
+                         "inputValue": 7.93,
+                         "outputValue": 340
+                         },
+                         {
+                         "inputValue": 8.446,
+                         "outputValue": 369.589
+                         }
+                         ],
+                         "offsets": [
+                         {
+                         "offset": 0.95
+                         }
+                         ]
+}
+  ]
   }')
   
   curves <- repgen:::readRatingCurves(curvesJson)
   
   expect_is(curves, 'data.frame')
   expect_equal(nrow(curves), 1)
-  expect_equal(curves[1,][['startOfPeriod']], '2015-10-06T16:06:01-05:00')
```

Comment:
I am now calculating the startofPeriod and endOf Period in the parseTSSratingCurves so i removed these two tests from here